optuna/study/_optimize.py

```diff
@@ -217,111 +211,42 @@ def _run_trial(
         state = TrialState.FAIL
         func_err = e
         func_err_fail_exc_info = sys.exc_info()
-    else:
-        # TODO(hvy): Avoid checking the values both here and inside `Study.tell`.
-        values, values_conversion_failure_message = _check_and_convert_to_values(
-            len(study.directions), value_or_values, trial.number
-        )
-        if values_conversion_failure_message is not None:
-            state = TrialState.FAIL
-        else:
-            state = TrialState.COMPLETE
 
     if study._storage.is_heartbeat_enabled():
         assert stop_event is not None
         assert thread is not None
         stop_event.set()
         thread.join()
 
-    # `Study.tell` may raise during trial post-processing.
+    # `Study._tell` may raise during trial post-processing.
     try:
-        study.tell(trial, values=values, state=state)
+        # Note: call `Study._tell` instead of `Study.tell` to suppress warning.
```

Comment:
Or, we can omit this line altogether. Code is self explanatory.
```suggestion
```

Reply:
> Code is self explanatory.

Agree after your comment. I removed the comment.